### PR TITLE
Add nonlinear Stommel demo

### PR DESCRIPTION
**Pull request by francispoulin**

### CODE REVIEW

**Comment by wence-** [comment]
Hi Francis, thanks for the sys.exit fix.  I looks like this pull request also introduces a new demo.  I will cherry-pick the small change and leave the larger here to be merged separately.

**Comment by francispoulin** [comment]
Hello,

Sorry that I combined those two updates.  That was clearly a bad idea and made things harder on you.  Will try and avoid that in the future and keep things simple.

Francis



------------------
Francis Poulin
Associate Professor
Department of Applied Mathematics
University of Waterloo

email:           fpoulin@uwaterloo.ca
Web:            https://uwaterloo.ca/poulin-research-group/
Telephone:  +1 519 888 4567 x32637

________________________________

**Comment by francispoulin** [comment]
Hello,

I am writing to check in to see if there might be a problem with the wind-driven gyre demo, or maybe it's just all round business.

If there is something i can do to help please let me know.

Cheers, Francis


------------------
Francis Poulin
Associate Professor
Department of Applied Mathematics
University of Waterloo

email:           fpoulin@uwaterloo.ca
Web:            https://uwaterloo.ca/poulin-research-group/
Telephone:  +1 519 888 4567 x32637

________________________________

File: demos/nonlinear_QG_winddrivengyre/qg_refs.bib~ [hunk 1]
```diff
@@ -0,0 +1,31 @@
+@Book{Pedlosky:1992,
```
Comment (miklos1) [inline demos/nonlinear_QG_winddrivengyre/qg_refs.bib~]: `*~` file, probably shouldn't be here.

Comment (francispoulin) [inline demos/nonlinear_QG_winddrivengyre/qg_refs.bib~]: Sorry about that.  You are correct.

I just pushed it to github.  Do I need to do anything else at this stage?

**Comment by dham** [comment]
There are a large number of maths, notation, typographic and code bugs in this demo. I think I have fixed these in the new PR #1012

**Pull request closed without merging**